Libraries/dotNetRDF/Writing/HtmlWriter.cs

```diff
@@ -187,7 +187,9 @@ private void GenerateOutput(HtmlWriterContext context)
             context.HtmlWriter.RenderBeginTag(HtmlTextWriterTag.Tbody);
 
             TripleCollection triplesDone = new TripleCollection();
-            foreach (INode subj in context.Graph.Triples.SubjectNodes)
+            
+            // HACK: Getting subject nodes explicitly while BaseTripleCollection.SubjectNodes is broken for indexed graphs
+            foreach (INode subj in context.Graph.Triples.Select(t => t.Subject).Distinct())
```

Comment:
I can see why you do this as it is probably the only workaround for the in-memory indexed graph, but I'm concerned that the .Distinct() is not scalable for larger graphs (not so much the in-memory graphs, but ones where the triples might be materialized from some backing store).

I wonder if it might be better to have a look at fixing SubjectNodes instead.